### PR TITLE
Fix rendering of theme catalog page

### PR DESCRIPTION
**Pull request by mickaelandrieu**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | The page Customize > Design > Theme Catalog should works as expected
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | http://forge.prestashop.com/browse/BOOM-5674
| How to test?  | cf description

#### Important guidelines

* Make sure [your local branch is up to date](https://help.github.com/articles/syncing-a-fork/) before commiting your changes!
* Your code MUST respect [our Coding Standards](http://doc.prestashop.com/display/PS16/Coding+Standards) (for code written in PHP, JavaScript, HTML/CSS/Smarty/Twig, SQL)!

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/9255)

### CODE REVIEW

File: src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php [hunk 1]
```diff
@@ -43,21 +43,7 @@ class ThemeCatalogController extends FrameworkBundleAdminController
...
-
-        $pageContent = file_get_contents(
-            'https://addons.prestashop.com/iframe/search-' . $versionHelper->getMajorVersion() . '.php?'
-            . http_build_query([
-                'psVersion' => $versionHelper->getVersion(),
-                'isoLang' => $this->getContext()->language->iso_code,
-                'isoCurrency' => $this->getContext()->currency->iso_code,
-                'isoCountry' => $this->getContext()->country->iso_code,
-                'activity' => $configuration->getInt('PS_SHOP_ACTIVITY'),
-                'parentUrl' => $request->getSchemeAndHttpHost(),
```
Comment (mickaelandrieu) [inline src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php:57]: looks like a good idea, but url was malformed.

File: src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php [hunk 2]
```diff
@@ -71,4 +57,21 @@ public function indexAction(Request $request)
...
+    {
+        $psVersion = $this->get('prestashop.core.foundation.version')->getVersion();
+        $parent_domain = $request->getSchemeAndHttpHost();
+        $context = $this->getContext();
+        $currencyCode = $context->currency->iso_code;
+        $languageCode = $context->language->iso_code;
+        $countryCode = $context->country->iso_code;
+        $activity = $this->get('prestashop.adapter.legacy.configuration')->getInt('PS_SHOP_ACTIVITY');
+
+        return "https://addons.prestashop.com/iframe/search-1.7.php?psVersion=$psVersion&isoLang=$languageCode&isoCurrency=$currencyCode&isoCountry=$countryCode&activity=$activity&parentUrl=$parent_domain&onlyThemes=1";
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php]: Keep the http_build_query that use url encode, or use something more readable :/

Comment (mickaelandrieu) [reply src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php]: the http_build_query doesn't provide the right url, please look at comments :/ I've tried with urlencode this doesn't works too, addons server needs the raw/unparsed url.

File: src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php (hunk 1 above)
Comment (mickaelandrieu) [reply src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php:57]: for the record @PierreRambaud, Addons platform does't expect a well formated parameter value for `parentUrl`.

For instance, let's say parentUrl value is `https://your-awesome-website.com`, then you expect to send it this way:

```
https://addons.prestashop.com/iframe/search-17.php?parentUrl=https%3A%2F%2Fyour-awesome-website.com&...
```

But the **only** valid url which works for addons is this one:

```
https://addons.prestashop.com/iframe/search-17.php?parentUrl=https://your-awesome-website.com&...
```

This is why I've give up to use the correct way to generate url and RFC-compliant urls here.

I hope you'll better understand the issue and will validate this pull request :+1:

Comment (PierreRambaud) [reply src/PrestaShopBundle/Controller/Admin/Improve/Design/ThemeCatalogController.php:57]: Weird, really weird -_-'

**Review by PierreRambaud — APPROVED** [review]

**Comment by PierreRambaud** [comment]
Thanks @mickaelandrieu

**Pull request merged**